### PR TITLE
feat(nodefs): mmap and msync

### PR DESCRIPTION
**Pull request by petersalomonsen**
add implementation of mmap and msync for NODEFS

### CODE REVIEW

File: tests/fs/test_mmap.c [hunk 1]
```diff
@@ -176,7 +179,7 @@ int main() {
     printf("/yolo/sharedoffset.txt content=%s %d\n", buffer + offset, offset);
     fclose(fd);
   }
-
+#if !defined(NODEFS)
```
Comment (kripken) [inline tests/fs/test_mmap.c]: why is this part disabled?

File: src/library_nodefs.js [hunk 2]
```diff
@@ -292,6 +292,27 @@ mergeInto(LibraryManager.library, {
         }
 
         return position;
+      },
+      mmap: function(stream, buffer, offset, length, position, prot, flags) {
+        if (!FS.isFile(stream.node.mode)) {
+          throw new FS.ErrnoError({{{ cDefine('ENODEV') }}});
+        }
+        var ptr = _malloc(length);
+        NODEFS.stream_ops.read(stream, buffer, ptr + offset , length, position);
```
Comment (kripken) [inline src/library_nodefs.js]: extra space before one of the `,`s

File: src/library_nodefs.js [hunk 3]
```diff
@@ -292,6 +292,27 @@ mergeInto(LibraryManager.library, {
...
+
```
Comment (kripken) [inline src/library_nodefs.js]: comparing to MEMFS, `offset` is not used there, but you add it here - is that a bug there, or here? If we aren't sure, an assertion that the offset is 0 might be good enough for now.

File: tests/fs/test_mmap.c (hunk 1 above)
Comment (petersalomonsen) [reply tests/fs/test_mmap.c]: For `NODEFS` files are not kept in memory like for `MEMFS` and so there is no memory allocation when e.g. appending to files. So this scenario is not a case for `NODEFS`. For example the property `stream.node.usedBytes` does not exist for `NODEFS`

File: src/library_nodefs.js (hunk 3 above)
Comment (petersalomonsen) [reply src/library_nodefs.js]: not sure actually, so I've added the assertion as you suggest.

**Pull request merged**